libraries/GCS_MAVLink/GCS_Common.cpp

```diff
@@ -1713,6 +1713,45 @@ void GCS_MAVLINK::send_accelcal_vehicle_position(uint32_t position)
     }
 }
 
+
+float GCS_MAVLINK::vfr_hud_airspeed() const
+{
+    // because most vehicles don't have airspeed sensors, we return a
+    // different sort of speed estimate in the relevant field for
+    // comparison's sake.
+    return AP::gps().ground_speed();
```

Comment:
I thought about it, and didn't do it for the same reason I haven't pulled common fence-reporting code back into GCS_MAVLINK: we don't have a good way to excise that code from vehicles that don't use that sensor! (issue #8387)

If the Copter PR that adds AirSpeed support to Copter goes in (#8430), I'd definitely move the airspeed thing up.  While it's just Plane that has airspeed-sensor-support - not-so-much.

I agree with you on the "send 0" thing, however.

Reply:
> we don't have a good way to excise that code from vehicles that don't use that sensor

What's the issue with this exactly? The only method that would get built in is the `AP_Airpseed::get_singleton`, which, by the way, is already included because of https://github.com/ArduPilot/ardupilot/blob/master/libraries/GCS_MAVLink/GCS_Common.cpp#L1145